modules/pool/invest/components/PoolInvestProportional.tsx

```diff
@@ -100,10 +106,7 @@ export function PoolInvestProportional({ onShowPreview }: Props) {
                                     <Box flex="1">
                                         <TokenSelectInline
                                             tokenOptions={option.tokenOptions}
-                                            selectedAddress={
-                                                selectedOptions[`${option.poolTokenIndex}`] ||
-                                                option.tokenOptions[0].address
-                                            }
+                                            selectedAddress={selectedOptions[`${option.poolTokenIndex}`]}
```

Comment:
selectedOptions[`${option.poolTokenIndex}`] can be undefined when the token does not have multiple options, this does not look correct.

taken from above:
if (token.hasMultipleTokenOptions) {
    setSelectedOption(index, token.address);
}

Reply:
It could also be undefined on the first render

Reply:
hasMulitpleTokenOptions = option.tokenOptions.length > 1. and `TokenSelectInline` has the same condition so I don't think i can be undefined.

the ordinal code was `selectedAddress={selectedOptions[`${option.poolTokenIndex}`] || option.tokenOptions[0].address}` but that seemed to end up with other calculation side effects in useInvest because selectedOptions isn't set until the user actually clicks the dropdown.

I call setSelectedOption in the useEffect(). That isn't guaranteed to set on the 1st renders?

Reply:
if hasMulitpleTokenOptions === false
selectedOptions[`${option.poolTokenIndex}`] is undefined, no?

whereas 
selectedOptions[`${option.poolTokenIndex}`] || option.tokenOptions[0].address

guaranteed that an address was always being passed in.

Reply:
I guess my logic is that if hasMulitpleTokenOptions === false, then we shouldn't be rendering the TokenSelectInline, and therefore selectedOptions[${option.poolTokenIndex}] wouldn't be called? Maybe that's the flaw in my logic.

I can include `|| option.tokenOptions[0].address` so that we do not get an undefined.